test/e2e_node/critical_pod_test.go

```diff
@@ -78,15 +76,24 @@ var _ = framework.KubeDescribe("CriticalPod [Serial] [Disruptive] [NodeFeature:C
 					v1.ResourceMemory: resource.MustParse("100Mi"),
 				},
 			})
+
 			nonCriticalBestEffort := getTestPod(false, bestEffortPodName, v1.ResourceRequirements{})
+			capacity := getNodeCPUAndMemoryCapacity(f)
+			used := v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("200m"),
+				v1.ResourceMemory: resource.MustParse("200Mi"),
+			}
+			available := getAvailableCPUAndMemoryCapacity(capacity, used)
 			criticalPod := getTestPod(true, criticalPodName, v1.ResourceRequirements{
 				// request the entire resource capacity of the node, so that
 				// admitting this pod requires the other pod to be preempted
-				Requests: getNodeCPUAndMemoryCapacity(f),
+				Requests: available,
```

Comment:
The test should verify that a critical pod will be admitted and preempt other pods in case of lacking resources, so we have to do this in order to make the test make sense.